apps/explorer/src/utils/getAmout.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    getPaySuiTransaction,
+    getPayTransaction,
+    getTransferSuiTransaction,
+    getTransferObjectTransaction,
+    getTransactionKindName,
+} from '@mysten/sui.js';
+
+import type {
+    SuiTransactionKind,
+    TransactionEffects,
+    SuiEvent,
+} from '@mysten/sui.js';
+
+function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+    if (value === null || value === undefined) return false;
+    return true;
+}
+
+const getCoinType = (
+    txEffects: TransactionEffects,
+    address: string
+): string | null => {
+    const events = txEffects?.events || [];
+    const coinType = events
+        ?.map((event: SuiEvent) => {
+            const data = Object.values(event).find(
+                (itm) => itm?.owner?.AddressOwner === address
+            );
+            return data?.coinType;
+        })
+        .filter(notEmpty);
+    return coinType?.[0] ? coinType[0] : null;
+};
+
+type FormattedBalance = {
+    amount?: number | null;
+    coinType?: string | null;
+    isSuiCoin?: boolean;
+    recipientAddress: string;
+}[];
+
+export function getAmount(
+    txnData: SuiTransactionKind,
+    txnEffect?: TransactionEffects
+): FormattedBalance | null {
+    const txKindName = getTransactionKindName(txnData);
+    if (txKindName === 'TransferObject') {
+        const txn = getTransferObjectTransaction(txnData);
+        return txn?.recipient
+            ? [
+                  {
+                      recipientAddress: txn?.recipient,
+                  },
+              ]
+            : null;
+    }
+
+    if (txKindName === 'TransferSui') {
+        const txn = getTransferSuiTransaction(txnData);
+        return txn?.recipient
+            ? [
+                  {
+                      recipientAddress: txn.recipient,
+                      amount: txn?.amount,
+                      coinType:
+                          txnEffect && getCoinType(txnEffect, txn.recipient),
+                      isSuiCoin: true,
+                  },
+              ]
+            : null;
+    }
+
+    const paySuiData =
+        getPaySuiTransaction(txnData) ?? getPayTransaction(txnData);
+
+    const amountByRecipient = paySuiData?.recipients.reduce(
```

Comment:
It's confusing to me why we reduce into an object only to then pull just the values out of it

Reply:
The response for Pay and PaySui returns something like `{
   coins: SuiObjectRef[];
    recipients: string[];
    amounts: number[];
}` where it could be multipe coins, recipients or amounts. this just combines the coins for a given receiptent and match  amount and cointype